pkg/util/containerd/containerd_util.go

```diff
@@ -134,6 +138,61 @@ func (c *ContainerdUtil) GetEvents() containerd.EventService {
 	return c.cl.EventService()
 }
 
+// GetContainers interfaces with the containerd api to get the list of containers.
+func (c *ContainerdUtil) GetContainers() ([]*cspec.Container, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.queryTimeout)
+	defer cancel()
+
+	dContainers, err := c.Containers()
+	if err != nil {
+		return nil, err
+	}
+
+	uContainers := make([]*cspec.Container, 0, len(dContainers))
+	for _, dContainer := range dContainers {
+		ctxNamespace := namespaces.WithNamespace(ctx, c.namespace)
+
+		info, err := dContainer.Info(ctxNamespace)
+		if err != nil {
+			_ = log.Errorf("Error extracting containerd %v information: %v", "Info", err)
```

Comment:
Would be useful to have a bit more information about the container in these error logs so we can more easily debug issues. I'm also tempted to make the `status` optional, as in when those look-ups fail just submit the container without them. But I guess we don't need to do that right now